backend/src/libs/interfaces/repository.interface.ts

```diff
@@ -1,5 +1,6 @@
 interface IRepository<T = unknown> {
-  find(payload: unknown): Promise<T>;
+  find(id: number): Promise<T>;
+  findByEmail(email: string): Promise<T>;
```

Comment:
i think you should use id: unknown and email:unknown, check create method

Reply:
First of all you shouldn't add findByEmail into IRepository. About id I think it's fine all id in db are strings but we will see